erpnext/accounts/doctype/financial_report_template/test_financial_report_engine.py

```
@@ -1668,3 +1668,233 @@ def test_build_condition_field_validation(self):
+		"""
+		Verify that an account with prior-period balance but no current-period
+		movements shows closing balance equal to opening balance (not 0).
+		"""
+		# Create a template that selects our test equity account
+		test_rows = [
+			{
+				"reference_code": "EQUITY_TEST",
+				"display_name": "Test Equity Reserve",
+				"data_source": "Account Data",
+				"balance_type": "Closing Balance",
+				"calculation_formula": f'["account_name", "=", "_Test Equity Reserve"]',
```

Ruff (0.14.13)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)